tests/api.c

```diff
@@ -33265,7 +33265,753 @@ static void test_wolfSSL_EVP_PKEY_assign(void)
+
+        wolfSSL_EVP_CIPHER_CTX_init(ctx);
+        AssertIntEQ(EVP_CipherInit(ctx, init, key, iv, 1), WOLFSSL_SUCCESS);
+
+        AssertIntEQ(wolfSSL_EVP_CIPHER_CTX_iv_length(ctx), iv_lengths[i]);
+
+        EVP_CIPHER_CTX_free(ctx);
+    }
+
+    printf(resultFmt, passed);
+#endif
+}
+static void test_wolfSSL_EVP_CIPHER_CTX_key_length(void)
+{
+#if defined(OPENSSL_ALL) && !defined(NO_DES3)
+    byte key[AES_256_KEY_SIZE] = {0};
+    byte iv[AES_BLOCK_SIZE] = {0};
+
+    printf(testingFmt, "wolfSSL_EVP_CIPHER_CTX_key_length");
+
+    EVP_CIPHER_CTX *ctx = EVP_CIPHER_CTX_new();
+    const EVP_CIPHER *init = EVP_des_ede3_cbc();
+
+    wolfSSL_EVP_CIPHER_CTX_init(ctx);
+    AssertIntEQ(EVP_CipherInit(ctx, init, key, iv, 1), WOLFSSL_SUCCESS);
+
+    AssertIntEQ(wolfSSL_EVP_CIPHER_CTX_key_length(ctx), 24);
+
+    EVP_CIPHER_CTX_free(ctx);
+    printf(resultFmt, passed);
+#endif
+}
+static void test_wolfSSL_EVP_CIPHER_CTX_set_key_length(void)
+{
+#if defined(OPENSSL_ALL) && !defined(NO_DES3)
+    byte key[AES_256_KEY_SIZE] = {0};
+    byte iv[AES_BLOCK_SIZE] = {0};
+    int keylen;
+
+    printf(testingFmt, "wolfSSL_EVP_CIPHER_CTX_set_key_length");
+
+    EVP_CIPHER_CTX *ctx = EVP_CIPHER_CTX_new();
+    const EVP_CIPHER *init = EVP_des_ede3_cbc();
+
+    wolfSSL_EVP_CIPHER_CTX_init(ctx);
+    AssertIntEQ(EVP_CipherInit(ctx, init, key, iv, 1), WOLFSSL_SUCCESS);
+
+    keylen = wolfSSL_EVP_CIPHER_CTX_key_length(ctx);
+
+    AssertIntEQ(wolfSSL_EVP_CIPHER_CTX_set_key_length(ctx, keylen),
+                 WOLFSSL_SUCCESS);
+
+    EVP_CIPHER_CTX_free(ctx);
+    printf(resultFmt, passed);
+#endif
+}
+static void test_wolfSSL_EVP_CIPHER_CTX_set_iv(void)
+{
+#if defined(OPENSSL_ALL) && defined(HAVE_AESGCM) && !defined(NO_DES3) &&\
+   !defined(NO_DES3)
+    byte key[DES3_KEY_SIZE] = {0};
+    byte iv[DES_IV_SIZE] = {0};
+    int ivLen, keyLen;
+
+    printf(testingFmt, "wolfSSL_EVP_CIPHER_CTX_set_iv");
+
+    EVP_CIPHER_CTX *ctx = EVP_CIPHER_CTX_new();
+    const EVP_CIPHER *init = EVP_des_ede3_cbc();
+
+    wolfSSL_EVP_CIPHER_CTX_init(ctx);
+    AssertIntEQ(EVP_CipherInit(ctx, init, key, iv, 1), WOLFSSL_SUCCESS);
+
+    ivLen = wolfSSL_EVP_CIPHER_CTX_iv_length(ctx);
+    keyLen = wolfSSL_EVP_CIPHER_CTX_key_length(ctx);
+
+    /* Bad cases */
+    AssertIntEQ(wolfSSL_EVP_CIPHER_CTX_set_iv(NULL, iv, ivLen), WOLFSSL_FAILURE);
+    AssertIntEQ(wolfSSL_EVP_CIPHER_CTX_set_iv(ctx, NULL, ivLen), WOLFSSL_FAILURE);
+    AssertIntEQ(wolfSSL_EVP_CIPHER_CTX_set_iv(ctx, iv, 0), WOLFSSL_FAILURE);
+    AssertIntEQ(wolfSSL_EVP_CIPHER_CTX_set_iv(NULL, NULL, 0), WOLFSSL_FAILURE);
+    AssertIntEQ(wolfSSL_EVP_CIPHER_CTX_set_iv(ctx, iv, keyLen), WOLFSSL_FAILURE);
+
+    /* Good case */
+    AssertIntEQ(wolfSSL_EVP_CIPHER_CTX_set_iv(ctx, iv, ivLen), 1);
+
+
+    EVP_CIPHER_CTX_free(ctx);
+    printf(resultFmt, passed);
+#endif
+}
+static void test_wolfSSL_EVP_PKEY_CTX_new_id(void)
+{
+#if defined(OPENSSL_ALL)
+    WOLFSSL_ENGINE* e = 0;
```

Comment:
To match consistency in the library, can you initialize your WOLFSSL_ENGINE pointer to NULL instead of 0?